### ISSUE TITLE
assignRole sometimes doesn't work

### ISSUE THREAD

**Issue by mgsmus** [spatie/laravel-permission#930]
This is my code:

```
public function register(PreRegisterRequest $request)
{
    $data = $request->validated() + [
            'active' => User::ACTIVE,
            'membership_status' => User::MEMBERSHIP_WAITING_FOR_APPROVAL,
        ];

    $data['password'] = bcrypt($data['password']);

    $customer = User::create($data);
    $customer->assignRole('customer');

    return redirect()->route('pre-register')->with('pre-register.success', 'Success');
}
```

The user is successfully being created but **sometimes** it doesn't insert record to model_has_roles table. But sometimes works. How can I debug this to find the error? There is no error in log files. I am having this problem once every 8-10 requests.

**Comment by drbyte** [comment]
@mgsmus I did some testing in a local app here and can't recreate this symptom.
I'm curious what you discover in your debugging.

**Comment by mgsmus** [comment]
@drbyte I think I found something. My laravel-permission version was 2.23.0 and I've updated to 2.24.0 a few days ago. It seems there is a new config entry in config/permission.php:

```
'column_names' => [
        /*
         * Change this if you want to name the related model primary key other than
         * `model_id`.
         *
         * For example, this would be nice if your primary keys are all UUIDs. In
         * that case, name this `model_uuid`.
         */
        'model_morph_key' => 'model_id',
    ],
```

I belive there is nothing wrong with assignRole().

**Comment by drbyte** [comment]
This package should be auto-merging missing configs (in memory, not into your config file), if not found in your config file. 

Are you saying that if that config entry is missing then your problems exist, but putting it in resolves the matter?

**Comment by mgsmus** [comment]
The config entry was missing in my config/permission.php. I don't know how exactly morphToMany works but $parentKey was always null [at this line](https://github.com/spatie/laravel-permission/blob/2a4e6fb31122a0c9974f45cbd6a67808f704155a/src/Traits/HasRoles.php#L46) because of the missing entry and no default provided.

I updated to latest release and republished the config file. I started to run tests again. I can't tell if it will work but it seems related.
